Tests/PackageFingerprintTests/FilePackageFingerprintStorageTests.swift

```diff
@@ -0,0 +1,153 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import struct Foundation.URL
+@testable import PackageFingerprint
+import PackageModel
+import SPMTestSupport
+import TSCBasic
+import XCTest
+
+final class FilePackageFingerprintStorageTests: XCTestCase {
+    func testHappyCase() throws {
+        let mockFileSystem = InMemoryFileSystem()
+        let storage = FilePackageFingerprintStorage(customFileSystem: mockFileSystem)
+        let registryURL = Foundation.URL(string: "https://example.packages.com")!
+        let sourceControlURL = Foundation.URL(string: "https://example.com/mona/LinkedList.git")!
+
+        // Add fingerprints for mona.LinkedList
+        let package = PackageIdentity.plain("mona.LinkedList")
+        try tsc_await { callback in storage.put(package: package, version: Version("1.0.0"),
+                                                fingerprint: .init(origin: .registry(registryURL), value: "checksum-1.0.0"),
+                                                callback: callback) }
+        try tsc_await { callback in storage.put(package: package, version: Version("1.0.0"),
+                                                fingerprint: .init(origin: .sourceControl(sourceControlURL), value: "gitHash-1.0.0"),
+                                                callback: callback) }
+        try tsc_await { callback in storage.put(package: package, version: Version("1.1.0"),
+                                                fingerprint: .init(origin: .registry(registryURL), value: "checksum-1.1.0"),
+                                                callback: callback) }
+        // Fingerprint for another package
+        let otherPackage = PackageIdentity.plain("other.LinkedList")
+        try tsc_await { callback in storage.put(package: otherPackage, version: Version("1.0.0"),
```

Comment:
consider creating private extension with blocking versions to simply these call sites